### ISSUE TITLE
.NET CLR Perf counters not to appinsights / Confused about guidance for Classic Cloud Services

### ISSUE THREAD

**Issue by milodotnet** [microsoft/ApplicationInsights-Home#474]
I'm really struggling with application insights and cloud services at the moment

The issue I'm trying to solve is that no matter what I do for some reason the .NET CLR related performance counter categories are not getting sent from my cloud service to application insights. I believe it is failing with a "category does not exist" error when it's attempting to do this from the code.

I can confirm the counters exists, and they even appear in table storage using the normal Azure Monitor / WadCfg config.

In trying to solve this I've come across alot of very old and out of date information and samples that don't do what they say they do.

Is there a "current" correct way to configure application insights for cloud service
- from code
- codeless (I'm confused as the the relationship between the WadCfg configured agent, and the agent installed in the sample in this repo)

If anything I would just love it if someone could point me to a working repo using application insights with a classic cloud service that actually has counters from the .NET CLR category appearing in app insights.

**Comment by TimothyMothra** [comment]
@cijothomas please take a look

**Comment by cijothomas** [comment]
@milodotnet What is the error message you are getting when trying to get the .net clr counter? Also can you paste the ai.config content where the additional counters are configured?

Something like the following is added to ai.config?
```
<Add PerformanceCounter="\.NET CLR Memory(??APP_CLR_PROC??)\# Bytes in all Heaps" ReportAs="Bytes in all Heaps" />
```

**Comment by milodotnet** [comment]
"category does not exist" seems to be error that get's reported. 

As per the documentation I should not need to add the counter explicitly to the ai config as CLR exceptions is a default counter that get's collected. But I have tried adding as per your example in the ai.config and in code.

I've also tried configuring the diagnostics extension via wadcfg to push to appinsights, but this doesn't seem to work at all for performance counters (they only go to table storage)

Is there a current and working example of this somewhere?

**Comment by cijothomas** [comment]
Are you seeing other default counters and only CLR excpetions is the one missing?

**Comment by milodotnet** [comment]
Yes, other counters appear, it's just the .NET CLR ones that don't work

**Comment by cijothomas** [comment]
@milodotnet Can you attempt to read the particular counter from manual code yourself, from the same app where application insights is not able to read it? 
This can't be a permission issue as other counters are working fine. Its possible that one counter got corrupted, but this would likely mean other means to read the same would also fail. Attempting to read the counter from the same app manually will help us narrow down further.

**Comment by milodotnet** [comment]
@cijothomas Yes, I have tried this as well, hence the reason I mentioned the "category does not exist" error. 

I've even tried startup tasks to run lodctr  on startup and using the wmi command, non of these help.

Do you have a sample that you can confirm works that I can use as reference?

**Comment by cijothomas** [comment]
Sorry I didn't quite follow:
Did you write own code to read the perf counter, and its also failing when run inside the same app?
If its failing, then the issue is with perf counter infra - sometimes counters get corrupted. (You can search about how to repair counters)

I don't know what the use of a sample - there is nothing wrong with your AI config as other counters are flowing.

**Comment by milodotnet** [comment]
@cijothomas 

Yes, I have tried writing code to access them from within the same process, and it can't access them due to the category does not exist issue.

As mentioned I've used various suggested approaches to "repairing the counters" as well and none of them have worked.

I've also tried re-imaging the cloud service instance and this hasn't done anything.

The reason I'm asking for a sample/docs is that I would like an example that someone can verify works for them to refer to. It's really difficult/frustrating at the moment as in addition to this specific problem there are various samples/approaches floating around as to how to currently get perf counters from a cloud service to application insights, and I'm not sure if there is consistent guidance as to how to do this at the current time since a lot of it is really old.

Currently it appears that if I follow what documentation I can cobble together that for an azure cloud service, you will never get the CLR counters into appinsights from a process, even though the documentation indicates you would. 

So I guess I would like to determine if the documentation is wrong, or if there is something else I'm suppose to do to make this work with cloud services.

Hope the clarifies what I'm after?

Thanks

**Comment by cijothomas** [comment]
If even after reimaging CloudService instance, one cannot read CLR counters from it - then its something to follow up with CloudService team. I don't think application insights can do anything other than reporting that counter reading failed.

This official docs are the only docs about cloud services.
https://docs.microsoft.com/en-us/azure/azure-monitor/app/cloudservices#performance-counters

If its proven that one cannot read clr counters from CloudService instances, then the above doc and product should be corrected to omit the CLR counter.
